bin/propolis-server/src/lib/initializer.rs

```diff
@@ -994,6 +1001,121 @@ impl<'a> MachineInitializer<'a> {
         smb_tables.commit()
     }
 
+    fn generate_bootorder(&self) -> Result<Option<Entry>, Error> {
+        eprintln!(
+            "generating bootorder with order: {:?}",
+            self.boot_order.as_ref()
+        );
+        let Some(boot_names) = self.boot_order.as_ref() else {
+            return Ok(None);
+        };
+
+        let mut order = fwcfg::formats::BootOrder::new();
+
+        let parse_bdf =
```

Comment:
rather than unwrapping here, i'd rather make `PciPath` implement `Into<Bdf>` (instead of `TryInto`), and pair it with an exhaustive test the the conversion is OK. we're doing the same checks in different words, for the two types. i'll do that as another followup.